File: workbench/pygmentex/coder_test.py
#! \usr\bin\env python
'''!latex:
\begin{coder}{Embedded comment}
This is a python function
\end{coder}
'''
def factorial(n):
  '''Compute n!'''
  if n > 1:
    return n * factorial(n-1)
  return 1

if __name__ == "__main__":
  """!latex:
Execute only if run as a script
  """
  print(f'factorial(10)={factorial(10)}')

File: workbench/pygmentex/coder_test.c
/*!latex:
\begin{coder}{Embedded comment}
This is a C program to compute 10!.
\end{coder}
*/
#include <stdio.h>
int main(int argc, char **argv) {
  int factorial = 1;
  for (int i=2;i<=10;i++) {
    factorial *= i;
  }
  printf("factorial(10)=%d\n",factorial);
  return 0;
}
